lib/Doctrine/DBAL/Migrations/Migration.php

```diff
@@ -95,14 +95,18 @@ public function writeSqlFile($path, $to = null)
 
         $this->outputWriter->write(sprintf("-- Migrating from %s to %s\n", $from, $to));
 
-        $sqlWriter = new SqlFileWriter(
+        return $this->createSqlFileWriter($path)
+                    ->write($sql, $direction);
+    }
+
+    protected function createSqlFileWriter(string $path): SqlFileWriter
```

Comment:
Ok, I'll go down the rabbit hole on this one. However it would probably not be only a test change.